### ISSUE TITLE
UnusedPrivateMember false positive with operator `in`

### ISSUE THREAD

**Issue by BraisGabin** [detekt/detekt#4242]
## Expected Behavior

```kotlin
val isInside = "bar" in listOf("foo".toRegex())

private operator fun Iterable<Regex>.contains(a: String): Boolean {
    return any { it.matches(a) }
}
```

The extension function should not be flagged because it's used.

## Context
You can look at #4221. I found the issue there. In this commit: https://github.com/detekt/detekt/pull/4241/commits/59f97d9f92b74b22a3b94f8b4690235c8f616124

## Your Environment
* Version of detekt used: main - 1.19.0-RC1

**Comment by BraisGabin** [comment]
Reopen because it seems that it wasn't fixed yet. This PR proves it and provides code to reproduce it: #4249

**Comment by chao2zhang** [comment]
The fix was effective for type resolution. The `operator` -> `name` match is probably broken for non-tr, including `contains` -> `in` and `plus` -> `+` etc.

**Comment by BraisGabin** [comment]
So, should we skip all the functions called `contains` for "plain detekt"? Or we should disable this rule for "plain detekt".
